classes/queued_table.php

```diff
@@ -0,0 +1,129 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace local_clonecategory;
+
+use table_sql;
+
+defined('MOODLE_INTERNAL') || die;
+require_once($CFG->libdir . '/tablelib.php');
+
+/**
+ * Clone category queued clones table.
+ *
+ * @package    local_clonecategory
+ * @copyright  2023 Matthew Hilton <matthewhilton@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class queued_table extends table_sql {
```

Comment:
I think a small but very useful change would be when you queue up a batch of category id1 to id2 you redirect to the report page but with two optional params for the from and to ids. Then it will filter to just the batch you did. If you navigate to that report page then you'll get the lot.

Reply:
Actually the above isn't right because the source is tasks which are temporary and so the report will shrink. If this does a join on the courses inside the $from then it can get a persistent list of the courses that should exist, and then show that as an unchanging table of progress and then show if it is queued vs running vs existing vs missing.

Reply:
IMO tying to join the two with sql is kinda hacky since the data is stored in the json customdata string on the adhoc tasks table, and then the other weird encoding on the logstore table.

I went a slightly different route and added a unique id to each form submission which is used to link the two. Then I just reuse the history table but filter by this id:

![Screenshot 2023-05-30 at 08-00-40 UNKN Home Administration Plugins Local plugins Clone Category Clone Category](https://github.com/catalyst/clonecategory/assets/17095477/e90b328c-d7f1-424f-b488-27efcdde0ca6)